addons/stock/wizard/stock_request_count.py

```diff
@@ -18,10 +19,30 @@ class StockRequestCount(models.TransientModel):
 
     def action_request_count(self):
         for count_request in self:
+            quants_to_count = count_request._get_quants_to_count()
             if count_request.set_count == 'set':
-                count_request.quant_ids.filtered(lambda q: not q.inventory_quantity_set).action_set_inventory_quantity()
-            count_request.quant_ids.with_context(inventory_mode=True).write(
-                count_request._get_values_to_write())
+                quants_to_count.filtered(lambda q: not q.inventory_quantity_set).action_set_inventory_quantity()
+            values = count_request._get_values_to_write()
+            quants_to_count.with_context(inventory_mode=True).write(values)
+
+    def _get_quants_to_count(self):
+        self.ensure_one()
+        quants_to_count = self.quant_ids
+        if not self.env.user.has_group('stock.group_production_lot'):
+            return quants_to_count
+        domain = []
+        # Searches sibling quants for tracked product.
+        for quant in self.quant_ids:
+            if quant.product_id.tracking == 'none':
+                continue
+            quant_domain = ['&',
+                            ('product_id', '=', quant.product_id.id),
+                            ('location_id', '=', quant.location_id.id)]
+            domain = expression.OR([domain, quant_domain]) if domain else quant_domain
+        if domain:
+            sibling_quants = self.env['stock.quant'].search(domain)
+            quants_to_count |= sibling_quants
+        return quants_to_count
```

Comment:
use a set to create a domain without duplicates

Reply:
Okay got it, changes are done.

Reply:
@kasp-odoo That wasn't what is wanted... The domain itself is always a list, the set was for the location ids so we avoid duplicates in the domain (avoid to have multiple domain parts looking for the same product in the same location)
Anyway, I'll do it my side